exercises/binary-search-tree/example.h

```diff
@@ -0,0 +1,226 @@
+#if !defined(_BINARY_SEARCH_TREE_H_)
+#define _BINARY_SEARCH_TREE_H_
+
+#include <memory>
+#include <cstdint>
+
+namespace binary_tree
+{
+    template<typename T>
+    class binary_tree final
+    {
+    public:
+        class binary_tree_iter;
+        using binary_tree_ptr = std::unique_ptr<binary_tree>;
+
+        explicit binary_tree(T data)
+            : _data(std::move(data)),
```

Comment:
I don't think the data needs to be explicitly moved here. It's not a reference, so I'm almost certain the compiler figures that out anyways.

Reply:
On second thought, this is just an unnecessary fix, don't bother.